x/issuance/types/genesis_test.go

```diff
@@ -0,0 +1,142 @@
+package types_test
+
+import (
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/suite"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/kava-labs/kava/app"
+	"github.com/kava-labs/kava/x/issuance/types"
+)
+
+type GenesisTestSuite struct {
+	suite.Suite
+
+	addrs []sdk.AccAddress
+}
+
+func (suite *GenesisTestSuite) SetupTest() {
+	config := sdk.GetConfig()
+	app.SetBech32AddressPrefixes(config)
+
+	_, addrs := app.GeneratePrivKeyAddressPairs(2)
+	suite.addrs = addrs
+}
+
+func (suite *GenesisTestSuite) TestValidate() {
+	type args struct {
+		assets types.Assets
+	}
+	type errArgs struct {
+		expectPass bool
+		contains   string
+	}
+	testCases := []struct {
+		name    string
+		args    args
+		errArgs errArgs
+	}{
+		{
+			"default",
+			args{
+				assets: types.DefaultAssets,
+			},
+			errArgs{
+				expectPass: true,
+				contains:   "",
+			},
+		},
+		{
+			"with asset",
+			args{
+				assets: types.Assets{
+					types.NewAsset(suite.addrs[0], "usdtoken", []sdk.AccAddress{suite.addrs[1]}, false),
+				},
+			},
+			errArgs{
+				expectPass: true,
+				contains:   "",
+			},
+		},
+		{
+			"blocked owner",
+			args{
+				assets: types.Assets{
+					types.NewAsset(suite.addrs[0], "usdtoken", []sdk.AccAddress{suite.addrs[0]}, false),
+				},
+			},
+			errArgs{
+				expectPass: false,
+				contains:   "asset owner cannot be blocked",
+			},
+		},
+		{
+			"empty owner",
+			args{
+				assets: types.Assets{
+					types.NewAsset(sdk.AccAddress{}, "usdtoken", []sdk.AccAddress{suite.addrs[0]}, false),
+				},
+			},
+			errArgs{
+				expectPass: false,
+				contains:   "owner must not be empty",
+			},
+		},
+		{
+			"empty blocked address",
+			args{
+				assets: types.Assets{
+					types.NewAsset(suite.addrs[0], "usdtoken", []sdk.AccAddress{sdk.AccAddress{}}, false),
+				},
+			},
+			errArgs{
+				expectPass: false,
+				contains:   "blocked address must not be empty",
+			},
+		},
+		{
+			"invalid denom",
+			args{
+				assets: types.Assets{
+					types.NewAsset(suite.addrs[0], "USD2T ", []sdk.AccAddress{}, false),
+				},
+			},
+			errArgs{
+				expectPass: false,
+				contains:   "invalid denom",
+			},
+		},
+		{
+			"duplicate asset",
+			args{
+				assets: types.Assets{
+					types.NewAsset(suite.addrs[0], "usdtoken", []sdk.AccAddress{suite.addrs[1]}, false),
+					types.NewAsset(suite.addrs[0], "usdtoken", []sdk.AccAddress{suite.addrs[1]}, false),
```

Comment:
add a test where only the denom is duplicated and the other fields are different.